### ISSUE TITLE
WebGL: U_ILLEGAL_ARGUMENT_ERROR with a minimal GDScript

### ISSUE THREAD

**Issue by TyounanMOTI** [godotengine/godot#71921]
### Godot version

4.0.dev (ba9491f87)

### System information

Windows 11 Chrome 109.0.5414.75 (Official Build), iOS 16.2 Safari

### Issue description

When exporting a simple project for Web with any GDScript like following:
```gdscript
extends Control

```
will print the following error to the console:
```
USER ERROR: U_ILLEGAL_ARGUMENT_ERROR
   at: _is_confusable (modules\text_server_adv\text_server_adv.cpp:6151)
```

![image](https://user-images.githubusercontent.com/173001/214089154-d88f3f44-b670-41ec-9743-30627677968e.png)

For more complex project, following additional errors are showing:
```
USER ERROR: U_FILE_ACCESS_ERROR
   at: _spoof_check (modules\text_server_adv\text_server_adv.cpp:6174)
USER ERROR: U_ILLEGAL_ARGUMENT_ERROR
   at: _is_confusable (modules\text_server_adv\text_server_adv.cpp:6151)
```
When I've inserted `print_line(p_string)` at text_server_adv.cpp line 6172, following logs are collected:
![image](https://user-images.githubusercontent.com/173001/214091164-cb5af630-56d2-446e-b1c4-dc20066565d5.png)

It seems that the identifiers of *.gd files are printed to the console.

### Steps to reproduce

1. Open minimal reproduction project.
2. Exports to Web.
3. Open exported site with Chrome and open developer console.
4. Following error logs will be seen (reproduced): `USER ERROR: U_ILLEGAL_ARGUMENT_ERROR`

### Minimal reproduction project

[text_server_adv_error.zip](https://github.com/godotengine/godot/files/10481582/text_server_adv_error.zip)

**Comment by bruvzg** [comment]
Export templates are build without ICU data, seems like if Unicode identifiers are checked at runtime (is it necessary, confusion checks are probably relevant to display warnings in the editor), so we can:
- include data by default (+4.5 MB)
- check if data is loaded and skip checks if it's not.

**Comment by TyounanMOTI** [comment]
I've confirmed this was fixed on Windows 11 Chrome and iOS 16.3 Safari. Thank you!
